src/Components/InputModel.re

```diff
@@ -14,26 +42,167 @@ let slice = (~start=0, ~stop=?, str) => {
 
 let removeBefore = (~count=1, index, text) => (
   slice(text, ~stop=index - count) ++ slice(text, ~start=index),
-  max(0, index - count),
+  max(0, index - count)
 );
 
 let removeAfter = (~count=1, index, text) => (
   slice(text, ~stop=index) ++ slice(text, ~start=index + count),
-  index,
+  index
 );
 
+
+let findNextWord = (text, cursorPosition) => {
+  let length = String.length(text)
+
+  let nextWord = ref(None:option(int));
+  nextWord := Some(-1)
+  String.sub(text, cursorPosition, length)
+    |> String.iteri((ind, cr) => switch (nextWord^) {
+      | Some(-1) => nextWord := String.contains(wordSeparators, cr) ? {Some(ind)} : {Some(-1)};
+      | None => nextWord := String.contains(wordSeparators, cr) ? {Some(ind)} : {Some(-1)};
+      | Some(_) => ()
+    });
+
+  let newIndex = switch (nextWord^) {
+   | Some(-1) => length
+   | Some(index) => index
+   | None => length
+   } ;
+
+  newIndex;
+}
```

Comment:
This code is really confusing to me. Why is there a distinction between `None` and `Some(-1)` when they seem to be treated exactly the same? Why is `nextWord` initialized with `None` and then immediately set to `Some(-1)`?

What is the difference between this and `findNextWordBoundary`? They seem very similar, except the default case is different. It seems like they could be combined into one function returning an `option`, deferring the decision of what to do with no match to the consumer.

Reply:
I'm sorry. This was an my attempt to implement `findNextWordBoundary` but it didn't workout. Will remove.

Reply:
No worries, that certainly does explain the unfinished look of it :D Worth noting that this would also have been caught if this had an interface file.

My suggestion to make the `find*WordBoundary` functions return an option instead still stands though. It's not obvious to me from the name what the default would be.